homeassistant/components/vodafone_station/sensor.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Any, Final
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory, UnitOfDataRate
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util.dt import utcnow
+
+from .const import _LOGGER, DOMAIN
+from .coordinator import VodafoneStationRouter
+
+NOT_AVAILABLE: list = ["", "N/A", "0.0.0.0"]
+
+
+@dataclass
+class VodafoneStationBaseEntityDescription:
+    """Vodafone Station entity base description."""
+
+    value: Callable[[Any, Any], Any] = lambda val, key: val[key]
+    is_suitable: Callable[[dict], bool] = lambda val: True
+
+
+@dataclass
+class VodafoneStationEntityDescription(
+    VodafoneStationBaseEntityDescription, SensorEntityDescription
+):
+    """Vodafone Station entity description."""
+
+
+def _calculate_uptime(value: dict, key: str) -> datetime:
+    """Calculate device uptime."""
+    d = int(value[key].split(":")[0])
+    h = int(value[key].split(":")[1])
+    m = int(value[key].split(":")[2])
+
+    return utcnow() - timedelta(days=d, hours=h, minutes=m)
+
+
+def _line_connection(value: dict, key: str) -> str:
+    """Identify line type."""
+
+    internet_ip = value[key]
+    dsl_ip = value.get("dsl_ipaddr")
+    fiber_ip = value.get("fiber_ipaddr")
+    internet_key_ip = value.get("vf_internet_key_ip_addr")
+
+    if internet_ip == dsl_ip:
+        return "xDSL"
+
+    if internet_ip == fiber_ip:
+        return "Fiber"
+
+    if internet_ip == internet_key_ip:
+        return "Internet Key"
+
+    return "Unknown"
+
+
+SENSOR_TYPES: Final = (
+    VodafoneStationEntityDescription(
+        key="wan_ip4_addr",
+        translation_key="external_ipv4",
+        icon="mdi:earth",
+        is_suitable=lambda info: info["wan_ip4_addr"] not in NOT_AVAILABLE,
+    ),
+    VodafoneStationEntityDescription(
+        key="wan_ip6_addr",
+        translation_key="external_ipv6",
+        icon="mdi:earth",
+        is_suitable=lambda info: info["wan_ip6_addr"] not in NOT_AVAILABLE,
+    ),
+    VodafoneStationEntityDescription(
+        key="vf_internet_key_ip_addr",
+        translation_key="external_ip_key",
+        icon="mdi:earth",
+        is_suitable=lambda info: info["vf_internet_key_ip_addr"] not in NOT_AVAILABLE,
+    ),
+    VodafoneStationEntityDescription(
+        key="inter_ip_address",
+        translation_key="active_connection",
+        icon="mdi:wan",
+        value=_line_connection,
+    ),
```

Comment:
You can make this device class enum. Make the values snake_case, and then you can add them to the translation files.

Reply:
Thx for the suggestion!